Foundation/NSNumberFormatter.swift

```diff
@@ -7,18 +7,34 @@
 // See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 //
 
-
+import CoreFoundation
 
 public class NSNumberFormatter : NSFormatter {
     
-    public override init() { NSUnimplemented() }
-    public required init?(coder aDecoder: NSCoder) {
-        NSUnimplemented()
+    typealias CFType = CFNumberFormatterRef
+    internal var __cfObject: CFType?
+    internal var _cfObject: CFType {
+        get {
+            if let obj = __cfObject {
+                return obj
+            } else {
+                #if os(OSX) || os(iOS)
+                    let numberStyle = CFNumberFormatterStyle(rawValue: CFIndex(self.numberStyle.rawValue))!
+                #else
+                    let numberStyle = CFNumberFormatterStyle(self.numberStyle.rawValue)
+                #endif
+                
+                let obj = CFNumberFormatterCreate(kCFAllocatorSystemDefault, locale._cfObject, numberStyle)
+                _setFormatterAttributes(obj)
+                __cfObject = obj
+                return obj
+            }
+        }
```

Comment:
I think it's probably the case that we want the _cfFormatter to only be used within NSDateFormatter itself; other internal-to-library clients should still use the public API (preferred) or a private swift method (if required for some reason).